src/obsidian_tears_backend/main.mo

```diff
@@ -460,6 +461,15 @@ actor class _ObsidianTearsBackend() = this {
     };
   };
 
+  // clean authTokenRegistry every 24 hours
+  system func timer(setGlobalTimer : Nat64 -> ()) : async () {
+    let secondsMore : Nat64 = 24 * 60 * 60;
+    let next = Nat64.fromIntWrap(Time.now()) + secondsMore;
```

Comment:
Unh, Time.now() returns nanoseconds, and then you are adding seconds. So instead of running every day, it will run a few seconds later.

More info: https://7po3j-syaaa-aaaal-qbqea-cai.ic0.app/base-library/ic-apis/time.html